src/packages/core/tests/server.test.ts

```diff
@@ -336,15 +336,71 @@ describe("server", () => {
 
       for (const specificPort of validPorts) {
         s = Ganache.server(defaultOptions);
+        assert.strictEqual(
+          s.address(),
+          null,
+          "server.address() should be null before `server.listen(...)` is called"
+        );
         await s.listen(<any>specificPort);
 
         try {
+          const address = s.address();
+          assert.notEqual(
+            address,
+            null,
+            "server.address() should not be null after listening"
+          );
+          assert.strictEqual(
+            address!.port,
+            +specificPort,
+            "server.address().port should be equal to the specified port after server.listen(nonZeroPort)"
+          );
           const req = request.post(`http://localhost:${+specificPort}`);
           await req.send(jsonRpcJson);
         } finally {
           await teardown();
         }
+        assert.strictEqual(
+          s.address(),
+          null,
+          "server.address() should be null after server is stopped"
+        );
+      }
+    });
+
+    it("finds a port to listen on when port is 0", async () => {
+      const zeroPort = 0;
+      s = Ganache.server(defaultOptions);
+      assert.strictEqual(
+        s.address(),
+        null,
+        "server.address() should be null before `server.listen(...)` is called"
+      );
+      await s.listen(zeroPort);
+
+      try {
+        const address = s.address();
+        assert.notEqual(
+          address,
+          null,
+          "server.address() should not be null after listening"
+        );
```

Comment:
Would it be worthwhile asserting that the value is a non-negative integer? Maybe not because we do ensure that sending a request there is successful 🤷